xdsl/ir/affine/affine_map.py

```diff
@@ -433,6 +437,26 @@ def is_projected_permutation(self, allow_zero_in_results: bool = False) -> bool:
         # Results are either dims or zeros and zeros can be mapped to input dims.
         return True
 
+    def apply_permutation(self, source: Sequence[_T]) -> tuple[_T, ...]:
+        """
+        Assert that `self` represents a permutation, and apply the permutation to
```

Comment:
```suggestion
        Assert that `self` represents a projected permutation, and apply the permutation to
```
Maybe this is MLIR (mis)terminology, but the example is not a permutation